authors/apps/comments/views.py

```diff
@@ -125,3 +129,55 @@ def get(self, request, id, *args, **kwargs):
         serializer = self.serializer_class(thread, many=True)
         return Response({"comments": serializer.data,
                          "thread count": thread.count()})
+
+
+class LikeCommentApiView(generics.ListCreateAPIView):
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = serializers.LikeCommentSerializer
+    renderer_classes = (LikeComementsJSONRenderer, )
+    queryset = LikeComment.objects.all()
+
+    def get_object(self, object, *args, **kwargs):
+        return get_object_or_404(
+            object.objects.all(),
+            id=self.kwargs.get('id')
+        )
+
+    def get_queryset(self, *args, **kwargs):
+        id = self.kwargs.get('id')
+        return self.queryset.filter(comment=id)
+
+    def post(self, request, *args, **kwargs):
+        id = self.kwargs.get('id')
+        comment = self.get_object(Comment)
+        like = {
+            'comment': comment.id,
+            'user': request.user.pk
+        }
+
+        if self.queryset.filter(comment=id, user=request.user.pk).count() >= 1:
+            raise ValidationError({
+                "message": "You already like this comment"
+            })
+        serializer = self.serializer_class(data=like, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status.HTTP_201_CREATED)
+
+
+class UnlikeCommentApiView(generics.RetrieveDestroyAPIView):
+
+    permission_classes = (IsAuthenticated, )
+    serializer_class = serializers.LikeCommentSerializer
+    queryset = LikeComment.objects.all()
+
+    def destroy(self, request, *args, **kwargs):
+        like = self.get_object()
+        if request.user.pk == like.user.id:
+            self.perform_destroy(like)
+            return Response({"message": "You have unliked this comment"})
+        else:
+            raise ValidationError({
+                "message": "You do not like this comment"
```

Comment:
This error message is not descriptive enough. When a person doesn't have permission to delete a liked comment, THis message should tell him that he does not have permission to delete the liked comment.

Reply:
@fahadmak The new by having one endpoint caters for this and therefore I have removed this entire validation.